bin/propolis-server/src/lib/vm/services.rs

```diff
@@ -51,16 +51,24 @@ impl VmServices {
         vm_properties: &InstanceProperties,
         ensure_options: &super::EnsureOptions,
     ) -> Self {
+        let vm_objects = vm_objects.lock_shared().await;
```

Comment:
If we do land this prior to https://github.com/oxidecomputer/propolis/pull/746, could we copy the interface identifiers here and pass them as owned values? Then we don't need to hold a read lock for the entire time we're setting up the oximeter producer, creating the kstats, etc.

Reply:
I'll bring in #746 to my branch and work with that.